source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,657 @@
 
-       echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
+      sudo systemctl daemon-reload
 
-#.  Exit root:
 
-    .. code-block:: bash
+#. Run the kubeadm command to initialize the master node:
 
-       exit
+   .. important:: 
 
-Configure and run Kubernetes
-****************************
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. 
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is fto use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
 
-This section describes how to configure and run Kubernetes with CRI-O and kata-runtime. We will be using  the :command:`kubectl` command-line interface to control Kubernetes.
+   .. code:: bash
 
-Configure and run CRI-O + kata-runtime
-======================================
+      sudo kubeadm init \
+      --pod-network-cidr 10.244.0.0/16 
 
-#.  Enable the CRI-O service:
 
-    .. code-block:: bash
+#. (Optional) Docker can be set to use the kata-runtime as the default Docker
+   runtime if desired. See the :ref:`kata` documentation for information on
+   how to do that.
 
-       sudo systemctl enable crio.service
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
+containerd
+----------
 
-    .. code-block:: bash
+For information on containerd as as Kubernetes runtime, see `What is
+containerd?`_. To use containerd as the Kubernetes container runtime:
 
-       sudo systemctl daemon-reload
-       sudo systemctl restart crio
+#. Start the containerd service and enable it to run at boot automatically:
 
+   .. code-block:: bash
 
+      sudo systemctl enable --now containerd.service
 
-#.  Initialize the master control plane with the command below and follow the displayed instructions to set up `kubectl`.  The :command:`init` command initializes a Kubernetes `control-plane node`_. If you will be adding a pod network add-on, you will need to add configuration to the :command:`init` command line. see the `Install pod network add-on`_ section for details.
 
-    .. code-block:: bash
+#. Configure kubelet to use containerd and reload the service.
 
-       sudo kubeadm init --cri-socket=/run/crio/crio.sock
+   .. code-block:: bash
 
-#.  Register kata-runtime as a `RuntimeClass handler`_:
+      sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
 
-    .. code-block:: bash
+      cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-containerd.conf
+      [Service]                                                 
+      Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+      EOF
 
-       cat << EOF | kubectl apply -f -
```

Comment:
It really would be better if we had a way to refer to the kata github for this.  It will be hard to keep this in sync.  The one ont he github is https://github.com/kata-containers/packaging/blob/master/kata-deploy/k8s-1.14/kata-qemu-runtimeClass.yaml but that refers to kata-qemu since it is part of kata-deploy.   This is why I think the setup scripts and these instructions should do the same thing.   Maybe, instead of using kata-containers in containers-virt your instructions should just instruct to use kata-deploy.  See https://github.com/kata-containers/packaging/tree/5717e6cbe77e370fd725919bd968dfb9582dea41/kata-deploy for more details.  They recommend kubectl apply -f https://raw.githubusercontent.com/kata-containers/packaging/master/kata-deploy/k8s-1.14/kata-qemu-runtimeClass.yaml  but I would actually look at the create_stack.sh script and follow the same process they use which sets up all the kata runtimes.  These instructions will be valid for longer if you do it that way.

Reply:
I'm not going to address it in this PR. See from OP
> There are still known complexities and issues with the process over all. This PR is not meant to address all issues or be the final set of instructions.

Reply:
We switched to kata-deploy as discussed. Let me know if there is anything else you want to add.